app/controllers/api/v1/bibs_controller.rb

```diff
@@ -1,20 +1,32 @@
 class Api::V1::BibsController < ApplicationController
+  # Receive teacher sets from a POST request.
+  # All records are inside params[:_json].
+  # Find or create a teacher set in the MLN db and its associated books.
   def create_or_update_teacher_sets
-    render status: 200, json: {
-      teacher_sets: [
-        {
-          id: 1,
-          title: 'Example title 1'
-        },
-        {
-          id: 2,
-          title: 'Example title 2'
-        }
-      ]
-    }.to_json
+    return if !request_valid?('create_or_update_teacher_sets')
+
+    saved_teacher_sets = []
+    params[:_json].each do |teacher_set_record|
+      teacher_set = TeacherSet.where(bnumber: teacher_set_record[:id]).first_or_initialize
+      teacher_set.update_attributes(
+        title: teacher_set_record[:title]
+        # TODO: update more teacher_set attributes here once the API contract is complete
+      )
+      teacher_set.update_included_book_list(teacher_set_record)
+      saved_teacher_sets << teacher_set if teacher_set.save
+    end
+
+    # build saved_teacher_sets_json_array for the response body
+    saved_teacher_sets_json_array = []
+    saved_teacher_sets.each do |saved_ts|
+      saved_teacher_sets_json_array << { id: saved_ts.id, bnumber: saved_ts.bnumber, title: saved_ts.title }
```

Comment:
While sending this data back is great, the question becomes do we really need it? If we're going log this information in the MLN logs anyway as well as send the response back to the lambda, I feel like it can be a bit redundant. I'm perfectly ok with a 200 response. Keep in mind, I'm already logging this information when I parse the records from the kinesis stream.